jupyter_server/services/kernels/handlers.py

```diff
@@ -140,6 +140,14 @@ def iopub_data_rate_limit(self):
     def rate_limit_window(self):
         return self.settings.get("rate_limit_window", 1.0)
 
+    @property
+    def subprotocol(self):
+        try:
+            protocol = self.selected_subprotocol
+        except Exception:
+            protocol = None
```

Comment:
Should this default to the [`kernel_ws_protocol`](https://github.com/jupyter-server/jupyter_server/blob/745f5ba3f00280c1e1900326a7e08463d48a3912/jupyter_server/serverapp.py#L1674-L1685) provided by the ServerApp?

Reply:
I'm not sure. We introduced this because Tornado doesn't support websocket subprotocols for some versions of websockets, and if the user provides a `kernel_ws_protocol` other than `''` (the default, legacy protocol), it means it must be in the list of the protocols given by the client when opening the websocket, but we couldn't receive that list because Tornado didn't support that... :smile: 
I think it's safer to fall back to the protocol we know doesn't need subprotocol negotiation.

Reply:
:+1: